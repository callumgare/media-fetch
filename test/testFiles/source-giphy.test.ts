import { expect } from "vitest";
import giphySource from "@/src/sources/giphy/index.js";
import {
  createBasicTestsForRequestHandlers,
  normaliseResponse,
} from "../testUtils.js";

createBasicTestsForRequestHandlers({
  source: giphySource,
  queries: {
    "single-media": {
      request: { id: "YsTs5ltWtEhnq" },
      checkResponse: (response) =>
        expect(normaliseResponse(response)).toMatchSnapshot(),
    },
    search: {
      request: { searchText: "happy" },
      checkResponse: (response) =>
        expect(response.media.length).toBeGreaterThan(5),
      numOfPagesToLoad: 2,
    },
  },
  queriesShared: {
    secrets: {
      apiKey: process.env.GIPHY_API_KEY,
    },
  },
});
