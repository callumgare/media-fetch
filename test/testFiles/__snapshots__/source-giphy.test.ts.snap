// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Can successfully get response from request handler "single-media" of source "giphy" 1`] = `
{
  "media": [
    {
      "dateUploaded": 2014-02-13T05:21:06.000Z,
      "files": [
        {
          "ext": "mp4",
          "fileSize": 358048,
          "height": 259,
          "image": false,
          "mimeType": "video/mp4",
          "type": "full",
          "url": "https://media4.giphy.com/media/YsTs5ltWtEhnq/giphy.mp4",
          "video": true,
          "width": 400,
        },
        {
          "ext": "mp4",
          "fileSize": 46036,
          "height": 98,
          "image": false,
          "mimeType": "video/mp4",
          "type": "thumbnail",
          "url": "https://media4.giphy.com/media/YsTs5ltWtEhnq/giphy-preview.mp4",
          "video": true,
          "width": 150,
        },
      ],
      "id": "YsTs5ltWtEhnq",
      "mediaFinderSource": "giphy",
      "title": "Confused Clark Kent GIF",
      "url": "https://giphy.com/gifs/confused-flying-YsTs5ltWtEhnq",
      "usernameOfUploader": "",
    },
  ],
  "request": {
    "id": "YsTs5ltWtEhnq",
    "queryType": "single-media",
    "source": "giphy",
  },
}
`;
